### PR TITLE
chore: pre-commit in CI

### PR DESCRIPTION
**Pull request by lmilbaum**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+repos:
```
Comment (yerdreic) [inline .pre-commit-config.yaml:1]: in general, not related to this line - do we need here to install pre-commit? https://pre-commit.com/#installation
(To add it to the dependencies) @lmilbaum

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
```
Comment (yerdreic) [inline .pre-commit-config.yaml:3]: why do we use specifically this revision?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - id: check-docstring-first
+      - id: requirements-txt-fixer
+      - id: check-merge-conflict
+      - id: no-commit-to-branch
+        args:
+          - "--branch"
+          - "main"
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: detect-aws-credentials
```
Comment (yerdreic) [inline .pre-commit-config.yaml:18]: what do we need lines 17+18 for?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        args:
+          - --allow-missing-credentials
+      - id: check-toml
```
Comment (yerdreic) [inline .pre-commit-config.yaml]: That is only for if we use toml files. right?

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      - id: check-merge-conflict
+      - id: no-commit-to-branch
+        args:
+          - "--branch"
+          - "main"
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: detect-aws-credentials
+        args:
+          - --allow-missing-credentials
```
Comment (yerdreic) [inline .pre-commit-config.yaml:20]: I see it means "Allow hook to pass when no credentials are detected" - What does that mean? and why do we want to use it?

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      - id: check-toml
+      - id: check-json
+  - repo: https://github.com/markdownlint/markdownlint
+    rev: v0.12.0
+    hooks:
+      - id: markdownlint
+        additional_dependencies: [rake]
```
Comment (yerdreic) [inline .pre-commit-config.yaml:25]: Does including this line here takes care of with he installation to rake for this hook?

**Review by yerdreic — COMMENTED** [review]
I have left some questions in the PR :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml:1]: It is required for running locally but not required for CI. Will update the README.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml:3]: It is the latest one

File: .pre-commit-config.yaml (hunk 3 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml:18]: It guard the contributor from sharing private keys and AWS creds.

File: .pre-commit-config.yaml (hunk 4 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml]: Right. Will remove it.

File: .pre-commit-config.yaml (hunk 5 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml:20]: Sharing creds (==secrets) is a known vulnerability. The purpose of this hook is to protect us from sharing them. Hoping that answers your question.

File: .pre-commit-config.yaml (hunk 6 above)
Comment (lmilbaum) [reply .pre-commit-config.yaml:25]: The repo takes care of listing its dependencies. pre-commit install them in its cache.

**Review by yerdreic — APPROVED** [review]

**Pull request merged**